app/src/main/java/net/squanchy/tweets/TweetsPageView.java

```diff
@@ -69,32 +72,42 @@ protected void onAttachedToWindow() {
         initList();
     }
 
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        if (disposable != null && !disposable.isDisposed()) {
```

Comment:
No need to do those checks (`dispose()` is idempotent, and if you get here without having ever subscribed, there is a bug and I'd rather crash early in that case